### PR TITLE
Rename Internal Packages

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

As an alternative to #33195, we should rename all of the `"private": true` packages with an `internal-` prefix. This makes it immediately clear whether a package is private or not at a glance.

### How to test the changes in this Pull Request:

1. Make sure CI passes, since it runs through all of the builds and tests, it should be considered comprehensive.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33218/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `JSON Style Fixes` 369417ecdbab00c281db7cc60ebb47ebfd5d8c7d
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: packages/js/internal-e2e-builds/project.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+     "root": "packages/js/internal-e2e-builds",
+     "sourceRoot": "packages/js/internal-e2e-builds",
+     "projectType": "library",
+     "targets": {
+     }
+ }
+
```
Comment (roykho) [inline packages/js/internal-e2e-builds/project.json]: New line at EOF

File: packages/js/internal-style-build/project.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+     "root": "packages/js/internal-style-build",
+     "sourceRoot": "packages/js/internal-style-build/src",
+     "projectType": "library",
+     "targets": {
+     }
+ }
```
Comment (roykho) [inline packages/js/internal-style-build/project.json]: New line at EOF

Comment (roykho) [inline packages/js/internal-style-build/project.json]: New line at EOF

**Review by roykho — DISMISSED** [review]
Just left some minor linting comments. Otherwise tested locally and build works. LGTM!

**Pull request merged**